sources/api/schnauzer/src/helpers.rs

```diff
@@ -1346,6 +1356,138 @@ handlebars_helper!(any_enabled: |arg: Value| {
     result
 });
 
+/// This helper writes out the default OCI runtime spec.
+///
+/// Default settings are specified in the following files:
+/// * sources/models/shared-defaults/oci-resource-limits.toml
+/// * sources/models/shared-defaults/oci-capabilities.toml
+
+fn get_param_path<'a>(helper: &'a Helper<'_, '_>, index: usize) -> Result<&'a String, RenderError> {
+    Ok(helper
+        .params()
+        .get(index)
+        .context(error::MissingParamSnafu {
+            index,
+            helper_name: helper.name(),
+        })?
+        .relative_path()
+        .context(error::MissingParamPathSnafu {
+            index,
+            helper_name: helper.name(),
+        })?)
+}
+
+pub fn oci_defaults(
+    helper: &Helper<'_, '_>,
+    _: &Handlebars,
+    _: &Context,
+    renderctx: &mut RenderContext<'_, '_>,
+    out: &mut dyn Output,
+) -> Result<(), RenderError> {
+    // To give context to our errors, get the template name (e.g. what file we are rendering), if available.
+    trace!("Starting oci_defaults helper");
+    let template_name = template_name(renderctx);
+    trace!("Template name: {}", &template_name);
+
+    // Check number of parameters, must be exactly two (OCI spec section to render and settings values for the section)
+    trace!("Number of params: {}", helper.params().len());
+    check_param_count(helper, template_name, 1)?;
+    trace!("params: {:?}", helper.params());
+
+    trace!("Getting the requested OCI spec section to render");
+    let oci_defaults_values = get_param(helper, 0)?;
+    // We want the settings path so we know which OCI spec section we are serializing.
+    let settings_path = get_param_path(helper, 0)?;
+    let oci_spec_section = settings_path
+        .split('.')
+        .last()
+        .expect("did not find (got None for) an oci_spec_section");
+
+    let result_lines = match oci_spec_section {
+        "capabilities" => oci_spec_capabilities(oci_defaults_values)?,
+        "resource-limits" => oci_spec_resource_limits(oci_defaults_values)?,
+        _ => panic!("bad oci_spec_section: {}", oci_spec_section),
+    };
+
+    // Write out the final values to the configuration file
+    out.write(result_lines.as_str())
+        .context(error::TemplateWriteSnafu {
+            template: template_name.to_owned(),
+        })?;
+
+    Ok(())
```

Comment:
can omit this if desired: and will need to remove question mark or semicolon or both on line 1416

Reply:
check for return type of `out.write()` -- if it is the same as the helperfunc, then deleting stuff is ok. Otherwise, keep Ok(())

Reply:
```
  #20 4.273 error[E0308]: mismatched types
  #20 4.273     --> api/schnauzer/src/helpers.rs:1410:5
  #20 4.273      |
  #20 4.273 1378 |   ) -> Result<(), RenderError> {
  #20 4.273      |        ----------------------- expected `std::result::Result<(), RenderError>` because of return type
  #20 4.273 ...
  #20 4.273 1410 | /     out.write(result_lines.as_str())
  #20 4.273 1411 | |         .context(error::TemplateWriteSnafu {
  #20 4.273 1412 | |             template: template_name.to_owned(),
  #20 4.273 1413 | |         })
  #20 4.273      | |__________^ expected struct `RenderError`, found enum `TemplateHelperError`
  #20 4.273      |
  #20 4.273      = note: expected enum `std::result::Result<_, RenderError>`
  #20 4.273                 found enum `std::result::Result<_, TemplateHelperError>`
 ```

Reply:
leaving Ok(()) intact